Updated the attested_tls sample to work with OpenSSL3+SymCrypt provider

justanotherminh opened the pull request:
Updated the attested_tls sample to work with OpenSSL3+SymCrypt provider

radhikaj commented on samples/attested_tls/README.md:
```diff
@@ -93,11 +93,11 @@ Note: Both of them can run on the same machine or separate machines.
   mkdir build
   cd build
-  cmake -DOE_CRYPTO_LIB=openssl_symcrypt_fips ..
+  cmake -DOE_CRYPTO_LIB=symcrypt_fips ..
```
I understand why you are trying to make this change, but I would like to keep this as is. Can you please keep this as is and add openssl3_symcrypt_fips as the new option?

justanotherminh replied:
This change makes it in line with the naming system used in the main codebase. It's also a more accurate description of what's going on because technically you can use the symcrypt engine with openssl 3 if deprecated support is enabled.

dcarpente commented:
Can you update test-samples.cmake to run attested_tls with symcrypt_prov_fips? That way the automated test can also verify that this sample is working with this configuration

dcarpente commented:
Maybe as a separate PR: it would also be great to update attested_tls Makefiles to build with symcrypt_prov_fips. Might require additions to files in pkgconfig.

Example change where I updated most of our Makefile samples to build with openssl 3:
https://github.com/openenclave/openenclave/commit/9e3a4f7232e421ddf01784bf273b10ccb9c009a9#diff-1715102d9823f79a30508e82f9237c03366c501a2e01bfa427b9cdeadd84a6fc

justanotherminh commented:
> Can you update test-samples.cmake to run attested_tls with symcrypt_prov_fips? That way the automated test can also verify that this sample is working with this configuration

I totally missed this. Thanks for pointing out Danny!

justanotherminh commented:
> Maybe as a separate PR: it would also be great to update attested_tls Makefiles to build with symcrypt_prov_fips. Might require additions to files in pkgconfig.
> 
> Example change where I updated most of our Makefile samples to build with openssl 3: [9e3a4f7#diff-1715102d9823f79a30508e82f9237c03366c501a2e01bfa427b9cdeadd84a6fc](https://github.com/openenclave/openenclave/commit/9e3a4f7232e421ddf01784bf273b10ccb9c009a9#diff-1715102d9823f79a30508e82f9237c03366c501a2e01bfa427b9cdeadd84a6fc)

I've added this to this PR

radhikaj commented on pkgconfig/README.md:
```diff
@@ -52,8 +52,9 @@ $ clang++-11 ${cxxflags} -o enc enc.cpp ${libs}
 - `mbedtls`
 - `openssl`
-- `openssl_symcrypt_fips`
+- `symcrypt_fips`
```
This is a breaking change. No one might be using openssl_symcrypt_fips, but nevertheless, it might be better to preserbe it.

radhikaj replied:
flexible to change my mind, if @mingweishih  chimes in otherwise.

mingweishih replied:
I agree. let's avoid this breaking change

justanotherminh replied:
Resolved

radhikaj reviewed (approved):
Approving with the caveat about the breaking change

mingweishih commented on pkgconfig/README.md:
```diff
@@ -52,8 +52,9 @@ $ clang++-11 ${cxxflags} -o enc enc.cpp ${libs}
-- `openssl_symcrypt_fips`
+- `symcrypt_fips`
 - `openssl_3`
+- `symcrypt_prov_fips`
```
can we do
`openssl_3_symcrypt_fips`
for consistency?

mingweishih replied:
In general, we want `OE_CRYPTO_LIB` to indicate what crypto library the sample uses. In this case, we need to say that it's openssl 3 backed by SymCrypt FIPS module. If we want to emphasis on the use of provider, then it can be `openssl_3_symcrypt_prov_fips`.

justanotherminh replied:
I made this change
